tests/integration_tests/datasources/test_nocloud.py

```diff
@@ -88,3 +88,38 @@ def test_nocloud_seedfrom_vendordata(client: IntegrationInstance):
     client.restart()
     assert client.execute("cloud-init status").ok
     assert "seeded_vendordata_test_file" in client.execute("ls /var/tmp")
+
+SMBIOS_USERDATA = """\
+#cloud-config
+runcmd:
+  - touch /var/tmp/smbios_test_file
+"""
+SMBIOS_SEED_DIR = "/smbios_seed"
+
+
+def setup_nocloud_serial(instance: LXDInstance):
+    subp(
+        [
+            "lxc",
+            "config",
+            "set",
+            instance.name,
+            "raw.qemu=-smbios "
+            f"type=1,serial=ds=nocloud;s=file://{SMBIOS_SEED_DIR};h=myhost",
+        ]
+    )
+
+
+@pytest.mark.lxd_setup.with_args(setup_nocloud_serial)
+@pytest.mark.lxd_use_exec
+@pytest.mark.skipif(
+    PLATFORM != "lxd_vm",
+    reason="Requires NoCloud with raw QEMU serial setup",
+)
+def test_smbios_seed(client: IntegrationInstance):
+    client.execute(f"mkdir {SMBIOS_SEED_DIR}")
+    client.write_to_file(f"{SMBIOS_SEED_DIR}/user-data", SMBIOS_USERDATA)
+    client.write_to_file(f"{SMBIOS_SEED_DIR }/smbios_seed/meta-data", "")
```

Comment:
This integration test fails due to the path typo duplicating SMBIOS_SEED_DIR/smbois_seed

```suggestion
    client.write_to_file(f"{SMBIOS_SEED_DIR}/meta-data", "")
```